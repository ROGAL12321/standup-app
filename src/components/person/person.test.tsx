import Person from "@/components/person";
import { render, screen, act } from "@testing-library/react";
import {
  statusName,
  mockedUser,
  mockedUserWithoutIssues,
  author,
  statusSummary,
} from "../../__mocks__/mocks";

describe("Person Component", () => {
  it("renders a person component", () => {
    render(
      <Person handleNextPerson={() => null} data={mockedUser} counter={60} />
    );

    const heading = screen.getByRole("heading");
    expect(heading).toHaveTextContent(author);
  });

  it("renders a person component with issues", () => {
    render(
      <Person handleNextPerson={() => null} data={mockedUser} counter={60} />
    );

    const issueStatus = screen.getByText(statusName);
    const issueSummary = screen.getByText(statusSummary);

    expect(issueStatus).toHaveTextContent(statusName);
    expect(issueSummary).toHaveTextContent(statusSummary);
  });

  it("renders a person component without issues", () => {
    render(
      <Person
        handleNextPerson={() => null}
        data={mockedUserWithoutIssues}
        counter={60}
      />
    );

    const noIssuesText = screen.getByText(/Nothing here/i);
    expect(noIssuesText).toBeInTheDocument();
  });

  it("renders a person component unchanged", () => {
    const { container } = render(
      <Person handleNextPerson={() => null} data={mockedUser} counter={60} />
    );

    expect(container).toMatchSnapshot();
  });
});
