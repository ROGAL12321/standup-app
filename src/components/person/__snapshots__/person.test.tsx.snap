// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person Component renders a person component unchanged 1`] = `
<div>
  <div
    class="personContainer"
  >
    <div
      class="nameContainer"
    >
      <img
        src="http://via.placeholder.com/48x48"
      />
      <h2>
        John Doe
      </h2>
      <span>
        60
        s
      </span>
    </div>
    <div
      class="issue"
    >
      <p
        class="issueTitle"
      >
         
        <img
          alt="Icon"
          src=""
        />
        <span>
          IN PROGRESS
        </span>
      </p>
      <p
        class="issueSummary"
      >
        some summary
      </p>
    </div>
    <div
      class="buttonContainer"
    >
      <button
        class="
    button
    centered
  "
      >
        Next Person
      </button>
    </div>
  </div>
</div>
`;
